internal/commands/exec/exec.go

```diff
@@ -86,6 +86,14 @@ func RunExec(config ExecConfig) (ret int) {
 			return 1
 		}
 		config.Logger.Trace("consul login complete")
+	} else {
+		consulConfig := &config.ConsulConfig
+		consulConfig.Namespace = config.GatewayConfig.Namespace
```

Comment:
I can add a comment here if y'all feel it necessary, but basically this is a bit of a hack to get our leaf cert setup working. Internally we're leveraging https://github.com/hashicorp/consul/tree/main/api/watch for grabbing and rotating leaf/root certs from the connect CA. It, unfortunately, has no concept of namespaces.

Since we're trying to match up the existence of a gateway based on the certificate it's pulled from connect, we need the leaf cert to be pulled using a Namespace parameter, this gets us that.